Code/Tools/SceneAPI/SceneData/Behaviors/ScriptProcessorRuleBehavior.cpp

```diff
@@ -377,11 +415,34 @@ namespace AZ::SceneAPI::Behaviors
             }
         };
 
-        [[maybe_unused]] Events::ProcessingResult fallbackResult;
-        if (LoadPython(context.GetScene(), scriptPath, fallbackResult))
+        EditorPythonConsoleNotificationHandler logger;
+        m_editorPythonEventsInterface->ExecuteWithLock(executeCallback);
+    }
+
+    bool ScriptProcessorRuleBehavior::DoPrepareForExport(Events::PreExportEventContext& context)
+    {
+        using namespace AzToolsFramework;
+
+        if (LoadPython())
+        {
+            auto defaultScript = FindMatchingDefaultScript(context.GetScene());
+            if (defaultScript)
+            {
+                SignalScriptForExportEvent(context, defaultScript.value());
+            }
+
+            [[maybe_unused]] Events::ProcessingResult fallbackResult;
+            auto manifestScript = FindManifestScript(context.GetScene(), fallbackResult);
+            if (manifestScript)
+            {
+                SignalScriptForExportEvent(context, manifestScript.value());
+            }
+        }
+        else
         {
-            EditorPythonConsoleNotificationHandler logger;
-            m_editorPythonEventsInterface->ExecuteWithLock(executeCallback);
+            AZ_Warning("scene", false,
```

Comment:
should this be an error instead of a warning? If python fails to start and the scene was using python in previous runs, you can get different product assets.

I don't think it should be a failure, because some processing can still be done.